daemon/mgr/spec_linux.go

```diff
@@ -55,3 +57,24 @@ func setupAppArmor(ctx context.Context, meta *ContainerMeta, spec *SpecWrapper)
 
 	return nil
 }
+
+func setupCapabilities(ctx context.Context, meta *ContainerMeta, spec *SpecWrapper) error {
+	var caplist []string
+	var err error
+
+	s := spec.s
+	if meta.HostConfig.Privileged {
+		caplist = caps.GetAllCapabilities()
+	} else {
+		caplist, err = caps.TweakCapabilities(s.Process.Capabilities.Effective, meta.HostConfig.CapAdd, meta.HostConfig.CapDrop)
+		if err != nil {
+			return err
+		}
+	}
+	s.Process.Capabilities.Effective = caplist
+	s.Process.Capabilities.Bounding = caplist
+	s.Process.Capabilities.Permitted = caplist
+	s.Process.Capabilities.Inheritable = caplist
```

Comment:
I think in this function, we do not need to always use `s.Process.Capabilities`. We can use `caps := s.Process. Capabilities` at the very beginning in the function. Since `Capabilities *LinuxCapabilities` is a pointer, then we do not worry about another copy or same address thing. And it will make code more clear.

Reply:
Agree :)